Add Sentinel governance proposal

axic opened the pull request.

axic commented:
Here is a demo implementation: https://github.com/ewasm/sentinel-governance

lrettig commented on sentinel.md:
```diff
@@ -7,4 +7,85 @@ This system contract performs two important tasks:
+    function getCost(CostKind kind) external view returns (uint64);
+
+    // Returns all the costs as an array, in order the enum is defined above.
+    function getAllCosts() etxernal view returns (uint64[]);
```
typo here in `external`

lrettig commented on sentinel.md:
```diff
@@ -7,4 +7,85 @@ This system contract performs two important tasks:
+}
+
+interface Sentinel {
+    // Takes a contract as an input, validates and injects metering statements and returns the upadted version.
```
typo here in "updated"

lrettig commented on sentinel.md:
```diff
@@ -7,4 +7,85 @@ This system contract performs two important tasks:
+interface Sentinel {
+    // Takes a contract as an input, validates and injects metering statements and returns the upadted version.
+    //
+    // Note, the Sentinel contract should also accept an incoming call with the `0x0061736D` selector and process
```
I don't understand this comment. Where does the `0x0061736D` selector come from?

axic replied:
Wasm header?

lrettig replied:
Aha :) Thanks

axic replied:
It is just a trick to avoid changing Hera at this moment 😉

lrettig commented on sentinel.md:
```diff
@@ -7,4 +7,85 @@ This system contract performs two important tasks:
+
+## Governance (upgradability)
+
+The Sentinel contract is split into two parts:
```
Might be worth mentioning the `SentinelGovernance` contract/interface here as well. It's a little confusing to say there are two parts, then introduce a third part further down.

lrettig commented on sentinel.md:
````diff
@@ -7,4 +7,85 @@ This system contract performs two important tasks:
+}
+```
+
+A sample implementation of the above exists under the [sentinel-governance](https://github.com/ewasm/sentinel-governance) project.
````
`SentinelGovernance` hasn't been implemented there yet, right?

lrettig commented:
This is awesome! Thanks @axic. Just reviewed and added some comments here.

axic commented:
@lrettig I think I've addressed all your comments. Please review again (including the entire file because it has changed in a different PR).

Would like to merge this to have a platform changing it with PRs.
